proto/regen/ecocredit/v1/tx.proto

```diff
@@ -273,12 +323,106 @@ message MsgCreateProjectResponse {
   string project_id = 1;
 }
 
+// MsgCreateUnregisteredProject is the Msg/CreateUnregisteredProject request type.
+message MsgCreateUnregisteredProject {
+  option (cosmos.msg.v1.signer) = "admin";
+
+  // admin is the address of the account creating the project that will become
+  // the admin of the project upon creation.
+  string admin = 1;
+
+  // metadata is any arbitrary string with a maximum length of 256 characters
+  // that includes or references metadata to attach to the project.
+  string metadata = 2;
+
+  // jurisdiction is the jurisdiction of the project. A jurisdiction has with
+  // the format: <country-code>[-<sub-national-code>[ <postal-code>]]
+  // The country-code must be 2 alphabetic characters, the sub-national-code
+  // can be 1-3 alphanumeric characters, and the postal-code can be up to 64
+  // alphanumeric characters. Only the country-code is required, while the
+  // sub-national-code and postal-code are optional and can be added for
+  // increased precision.
+  string jurisdiction = 3;
+
+  // reference_id is any arbitrary string used to reference the project with a
+  // maximum length of 32 characters.
+  string reference_id = 4;
+
+  // fee is the project creation fee. An equal fee is required if the project
+  // creation fee parameter is set. The provided fee can be greater than the
+  // parameter, but only the amount in the parameter will be charged.
+  //
+  // Since Revision 3
+  cosmos.base.v1beta1.Coin fee = 5;
+}
+
+// MsgCreateUnregisteredProjectResponse is the Msg/CreateUnregisteredProject response type.
+message MsgCreateUnregisteredProjectResponse {
+  // project_id is the unique identifier of the project.
+  string project_id = 1;
+}
+
+// MsgCreateOrUpdateApplication is the Msg/CreateOrUpdateApplication request type.
+message MsgCreateOrUpdateApplication {
```

Comment:
Why only one message for create and update? What fields can be updated after creation? This is different than Projects that have separate messages for creation and updating individual fields. These message make it very clear what can and cannot be updated. Their corresponding events also make it simple to subscribe to changes.

For applications it seems like only two fields should be updated: `metadata` and `withdraw`. It seems like you would make a new application instead of updating the project ID or credit class on an existing application. And for withdrawing, it isn't clear if that value can be updated multiple times. If you withdraw an application, would it make sense to "lock" the application state from any future updates and instead require creating a new application?

In this case, messages could be simplified to:
- MsgCreateApplication
- MsgUpdateApplicationMetadata
- MsgWithdrawApplication

Reply:
The main rationale is that there's a fair amount of ceremony for every new method - separate messages, validation, tests, etc. So I can split it up if the single method is overly complex. It's just a little more annoying for implementation.

Withdrawing would delete the application and you'd need to re-apply.

Reply:
Okay I can make separate messages. Not that much harder

Reply:
Is one message okay for update enrollment?

Reply:
Hmm yeah one message for update enrollment makes more sense. It would be common to update both the metadata and status at the same time. Although this would be the only ecocredit entity without its own dedicated update metadata msg :shrug:

Reply:
I think of the two options, separating out withdraw into it's own message makes the most sense. Like @paul121 said, the only two things we would really update are the `metadata` and `withdraw`. Having a separate withdraw message makes sense to me. Having a separate update metadata message I'm not sure about, mainly because I I don't know how often we would actually update application metadata or what that process would look like. If an issuer requested changes made to the application, would the project admin do we think it's important to have the issuer update the metadata to reflect the revised / new evidence?

Reply:
>  Having a separate update metadata message I'm not sure about, mainly because I I don't know how often we would actually update application metadata or what that process would look like. If an issuer requested changes made to the application, would the project admin do we think it's important to have the issuer update the metadata to reflect the revised / new evidence?

My understanding: When an issuer requests changes to an application I would expect the issuer to update the Enrollement status & metadata. I would then expect the project admin to update the Application metadata to address the requested changes. How revisions are represented is outside of the scope here, but could be included within the metadata content.